### PR TITLE
Let's Go [#127890783]

### PR DESCRIPTION
**Pull request by kdomanski**

### CODE REVIEW

File: build-go-tools.sh [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+mkdir -p binaries
+
+for tooldir in go/*; do
+  tool_name="$(basename $tooldir)"
+  echo "Building $tool_name"
+
+  echo " * Gliding up"
+  docker run -v "$HOME/bin/glide:/usr/bin/glide:ro" -v "$(readlink -f $tooldir):/go/src/$tool_name" -w "/go/src/$tool_name" -e GO15VENDOREXPERIMENT=1 golang:1.5 glide up
+  echo " * Building"
+  docker run -v "$HOME/bin/glide:/usr/bin/glide:ro" -v "$(readlink -f $tooldir):/go/src/$tool_name" -w "/go/src/$tool_name" -e GO15VENDOREXPERIMENT=1 golang:1.5 go build
```
Comment (colszowka) [inline build-go-tools.sh:19]: Wouldn't it rather make sense to first test and then build? My assumption would be that the test compiles the code anyway on the fly?

File: go/gitlab-network/main.go [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+	if len(os.Args) != 2 {
+		printUsage()
+		os.Exit(1)
+	}
+
+	switch os.Args[1] {
+	case "start":
+		err := createInterface(constInterfaceName)
+		if err != nil {
+			log.Fatalf("failed to create interface: %s\n", err.Error())
```
Comment (colszowka) [inline go/gitlab-network/main.go:158]: exit with non-zero?

File: go/gitlab-network/main.go [hunk 3]
```diff
@@ -0,0 +1,179 @@
...
+		}
+		break
+	case "stop":
+		err := tenus.DeleteLink(constInterfaceName)
+		if err != nil {
+			log.Fatalf("failed to delete interface: %s\n", err.Error())
```
Comment (colszowka) [inline go/gitlab-network/main.go:164]: here too?

File: go/gitlab-network/main.go [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+		}
+		break
+	case "show":
+		ip, err := getInterfaceIP(constInterfaceName)
+		if err != nil {
+			log.Fatalf("failed to get interface IP: %s\n", err.Error())
```
Comment (colszowka) [inline go/gitlab-network/main.go:170]: And here :)

**Comment by colszowka** [comment]
Looks good by me, but please also have a look @hinnerk 

What about unit tests?

File: build-go-tools.sh (hunk 1 above)
Comment (hinnerk) [inline build-go-tools.sh]: > Fatalf is equivalent to Printf() followed by a call to os.Exit(1).

https://golang.org/pkg/log/#Fatalf

File: go/gitlab-network/main.go (hunk 2 above)
Comment (hinnerk) [inline go/gitlab-network/main.go:158]: > Fatalf is equivalent to Printf() followed by a call to os.Exit(1).

https://golang.org/pkg/log/#Fatalf

**Pull request merged**